torchgeo/datasets/ssl4eo_benchmark.py

```diff
@@ -0,0 +1,366 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from matplotlib.colors import ListedColormap
+from torch import Tensor
+
+from .cdl import CDL
+from .geo import NonGeoDataset
+from .nlcd import NLCD
+from .utils import download_url, extract_archive
+
+
+class SSL4EOLBenchmark(NonGeoDataset):
+    """SSL4EO Landsat Benchmark Evaluation Dataset.
+
+    Dataset is intended to be used for evaluation of SSL techniques. Each
+    benchmark dataset consists of 25,000 images with corresponding land
+    cover classification masks.
+
+    Dataset format:
+
+    * Input landsat image and single channel mask
+    * 25,000 total samples split into train, val, test (70%, 15%, 15%)
+    * NLCD dataset version has 17 classes
+    * CDL dataset version has 134 classes
+
+    Each patch has the following properties:
+
+    * 264 x 264 pixels
+    * Resampled to 30 m resolution (7920 x 7920 m)
+    * Single multispectral GeoTIFF file
+
+    .. versionadded:: 0.5
+    """
+
+    url = "https://huggingface.co/datasets/torchgeo/{}/resolve/main/{}.tar.gz"
+
+    valid_input_sensors = ["tm_toa", "etm_toa", "etm_sr", "oli_tirs_toa", "oli_sr"]
+    valid_mask_products = ["cdl", "nlcd"]
+    valid_splits = ["train", "val", "test"]
+
+    image_root = "ssl4eo_l_{}_benchmark"
+    img_md5s = {
+        "tm_toa": "8e3c5bcd56d3780a442f1332013b8d15",
+        "etm_toa": "1b051c7fe4d61c581b341370c9e76f1f",
+        "etm_sr": "a33d5e3b9c309980ef15d453235843ed",
+        "oli_tirs_toa": "6e9d7cf0392e1de2cbdb39962ba591aa",
+        "oli_sr": "0700cd15cc2366fe68c2f8c02fa09a15",
+    }
+
+    mask_dir_dict = {
+        "tm_toa": "ssl4eo_l_tm_{}",
+        "etm_toa": "ssl4eo_l_etm_{}",
+        "etm_sr": "ssl4eo_l_etm_{}",
+        "oli_tirs_toa": "ssl4eo_l_oli_{}",
+        "oli_sr": "ssl4eo_l_oli_{}",
+    }
+    mask_md5s = {
+        "tm": {
+            "cdl": "3d676770ffb56c7e222a7192a652a846",
+            "nlcd": "261149d7614fcfdcb3be368eefa825c7",
+        },
+        "etm": {
+            "cdl": "dd2560b18b89dfe7f0e867fcf7217bd0",
+            "nlcd": "916f4a433df6c8abca15b45b60d005d3",
+        },
+        "oli": {
+            "cdl": "1cb057de6eafeca975deb35cb9fb036f",
+            "nlcd": "9de0d6d4d0b94313b80450f650813922",
+        },
+    }
+
+    year_dict = {
+        "tm_toa": 2011,
+        "etm_toa": 2019,
+        "etm_sr": 2019,
+        "oli_tirs_toa": 2019,
+        "oli_sr": 2019,
+    }
+
+    rgb_indices = {
+        "tm_toa": [2, 1, 0],
+        "etm_toa": [2, 1, 0],
+        "etm_sr": [2, 1, 0],
+        "oli_tirs_toa": [3, 2, 1],
+        "oli_sr": [3, 2, 1],
+    }
+
+    split_percentages = [0.7, 0.15, 0.15]
+
+    ordinal_label_map = {"nlcd": NLCD.ordinal_label_map, "cdl": CDL.ordinal_label_map}
+
+    cmaps = {"nlcd": NLCD.cmap, "cdl": CDL.cmap}
+
+    def __init__(
+        self,
+        root: str = "data",
+        input_sensor: str = "oli_sr",
+        mask_product: str = "cdl",
+        split: str = "train",
```

Comment:
In hindsight, I would have probably called these parameters `sensor` and `product` and removed the `split` parameter. We should probably do splitting in the data module to match our other datasets. But this can wait until 0.5, no rush to do it before the deadline.

Reply:
Also, I think we should put all datasets in one directory on HF to reduce the clutter.

Reply:
also post deadline or want to do it now?

Reply:
Post deadline